### PR TITLE
[coverage] Improves coverage

### PR DESCRIPTION
**Pull request by harman28**
* Refactors error class requires to handle unknown case
* Tests for method_missing and respond_to_missing
* Only Travis wears coveralls

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ rvm:
   - "2.4.0"
 before_install: gem update --system
 script:
+  - RUBY_ENV=travis
```
Comment (captn3m0) [inline .travis.yml]: You don't need this. You can use `TRAVIS=true` which already exists on all travis builds: https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables

File: test/test_helper.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
 require 'coveralls'
+require 'simplecov'
 require 'minitest/autorun'
 require 'webmock/minitest'
 require 'razorpay'
 
-Coveralls.wear!
+Coveralls.wear! if ENV['RUBY_ENV'] == 'travis'
```
Comment (captn3m0) [inline test/test_helper.rb]: Change this check as well. I'm concerned because coveralls didn't trigger for this PR

File: .travis.yml (hunk 1 above)
Comment (harman28) [inline .travis.yml]: Was hoping for something like that. Is it okay with I use `CI=true`?

Comment (captn3m0) [inline .travis.yml]: Yup, that works as well.

**Pull request merged**